Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py

```diff
@@ -524,6 +526,32 @@ def run_retry_on_rate_limit(self, args_for_next_run: dict):
                               scheduled_command=ScheduledCommand(command=demisto.command(), next_run_in_seconds=60,
                                                                  args=args_for_next_run))
 
+    def handle_error_with_metrics(self, res):
+        self.create_api_metrics(res.status_code)
+        err_msg = 'Error in API call [{}] - {}' .format(res.status_code, res.reason)
+        try:
+            # Try to parse json error response
+            error_entry = res.json()
+            err_msg += '\n{}'.format(json.dumps(error_entry))
+            raise DemistoException(err_msg, res=res)
+        except ValueError:
+            err_msg += '\n{}'.format(res.text)
+            raise DemistoException(err_msg, res=res)
+
+    def create_api_metrics(self, status_code):
+        execution_metrics = ExecutionMetrics()
+        ok_codes = (200, 201, 202, 204, 206)
+
+        if not execution_metrics.is_supported() or demisto.command() == 'test-module':
```

Comment:
`or demisto.command() == 'test-module':`
is this neccessary? Shouldnt we not have gotten here due to overwrite_rate_limit_retry? Better to keep the logic of whether or not we're running this flow in 1 place

Reply:
We didn't skip the test module in the retry mechanism, but you're right it makes more sense to skip them now. I'll add the overwrite_rate_limit_retry to them :)